### ISSUE TITLE
Add exceptBeforeSingleLine to lines-between-class-members

### ISSUE THREAD

**Issue by luxaritas** [eslint/eslint#11583]
**What rule do you want to change?**
lines-between-class-members

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option - exceptBeforeSingleLine 

**Please provide some example code that this change will affect:**

```js
class Foo{
  foobar(){}
  bar(){}

  baz(){
    // something
  }

  qux(){}
}
```

**What does the rule currently do for this code?**
"Always", "never", and either with "exceptAfterSingleLine" all throw an error.

**What will the rule do after it's changed?**
This will pass (given "always"). The idea is to "group" single-line members.

**Are you willing to submit a pull request to implement this change?**
Probably - looks doable on a rough glance

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal.

Personally, I'm not sure this option would make sense to add to ESLint core -- it seems like a fairly specific stylistic preference. We wouldn't be able to add individual options for every user's preference, so instead we generally only add an option to a rule if it would be useful for a fairly large percentage of our users, and in other cases we suggest that people [create a custom rule](https://eslint.org/docs/developer-guide/working-with-rules) instead. I'm not convinced this option would meet that bar, although I wouldn't object to adding it if another team member thinks it's worthwhile.
